test/distributed/test_c10d_ucc.py

```diff
@@ -1050,4 +1051,5 @@ def test_collectives(self):
         not torch.cuda._initialized
     ), "test_distributed must not have initialized CUDA context on main process"
 
-    run_tests()
+    if not TEST_WITH_ROCM:
```

Comment:
I think it might be better to use `if c10d.is_ucc_available():` instead, in case there are other setups in the future that don't have UCC backend.

Reply:
I have updated the skip logic to `not c10d.is_available() or not c10d.is_ucc_available()` and also apply the same change to NCCL and Gloo tests

Reply:
Looks good, thanks!